jetpack/migrate-v1-2-to-v1-3.md

```diff
@@ -0,0 +1,133 @@
+
+## Understanding the flows
+
+Before we discuss how to migrate from version 1.2 to 1.3, let's take a moment to understand the provisioning process of each version of the API.
+
+### Version 1.2 integration flow
+
+Below is a high-level diagram of the version 1.2 integration flow.
+
+![Jetpack version 1.2 flow diagram](/assets/jps-v1-2-overview.png)
+
+There are a few key things which we'll point out from this diagram:
+
+- A WordPress.com oAuth token from the user is required to call the endpoint
+  - This requires sending users through the WordPress oAuth flow
+  - This also requires that users have, or create, a WordPress.com account
+  - In some cases, the oAuth token may not be accepted if the site is already connected to another WordPress.com user
+- The host is required to set many values via WP-CLI or some other method to activate and setup Jetpack, Akismet, and VaultPress
+
+### Version 1.3 integration flow
+
+Below is a high-level diagram of the version 1.3 integration flow.
+
+![Jetpack version 1.3 flow diagram](/assets/jps-v1-3-overview.png)
+
+At first glance, you may notice that the diagram for version 1.3 is more complex. That's because version 1.3 of the API is definitely more complex beneath the hood. But, I'll point out that much of that complexity is hidden "under the hood".
+
+From the hosting partner's perspective, we believe the integration is actually much simpler for the following reasons:
+
+- oAuth is not required before calling the API
+- If a user does not already exist on WordPress.com then the API will attempt to create the user and connect the site automatically (if the partner is whitelisted)
+- The host does not have to run WP-CLI commands to install Akismet and VaultPress
+- The host does not have to run WP-CLI commands to set key for Jetpack, Akismet, and VaultPress
+
+We created version 1.3 with the goal of making the integration process simpler for hosting partners. Now, hosting partners should only have to:
+
+1. Make an API request to the `/v1.3/jpphp/provision` endpoint with the following information:
+   - Site URL
+   - User ID, email, or username
+   - The plan to provision to the site
+2. Choose whether to redirect the user based on the response from the API
+   - If the API returns a response that includes `auth_required: true`, then the user will need to connect their Jetpack site
+   - In this case, a `next_url` value should be returned, if possible, which will allow the user to connect their site to WordPress.com
+   - Alternatively, the user can click the "Set Up Jetpack" button within WordPress.com to go through the connection process and then receive their plan
+
+## Doing the migration
```

Comment:
This isn't major but a few of the below items are things that are new and required for v 1.3 and some things are things that are no longer required. It might make sense to split this out.

Reply:
Thanks for pointing that out. I think I'll go ahead and merge and we can consider that in the future.